fmn/rules/tracking_rules.py

```diff
@@ -36,35 +36,23 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.usernames = set(self._params)
 
-    def matches(self, message):
-        for package in message.packages:
-            if self.usernames & set(self._requester.get_package_owners(package)):
-                return True
-        for container in message.containers:
-            if self.usernames & set(self._requester.get_container_owners(container)):
-                return True
-        for module in message.modules:
-            if self.usernames & set(self._requester.get_module_owners(module)):
-                return True
-        for flatpak in message.flatpaks:
-            if self.usernames & set(self._requester.get_flatpak_owners(flatpak)):
-                return True
+    async def matches(self, message):
```

Comment:
Let's have a type hint here – what type of message would this have to be to ensure it has `packages`, `containers`, `modules`, `flatpaks` members?